airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/avro_parser.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from typing import Iterator, TextIO, BinaryIO, Union, Mapping
+from fastavro import reader
+import fastavro
+
+from .abstract_file_parser import AbstractFileParser
+from typing import Any
+
+
+# mapping from apache avro docs: https://avro.apache.org/docs/current/spec.html#schema_complex
+data_type_mapping = {
+    "null": "null",
+    "boolean": "boolean",
+    "int": "integer",
+    "long": "integer",
+    "float": "number",
+    "double": "number",
+    "bytes": "string",
+    "string": "string",
+    "record": "object",
+    "enum": "string",
+    "array": "array",
+    "map": "object",
+    "fixed": "string",
+}
+
+
+class AvroParser(AbstractFileParser):
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def is_binary(self) -> bool:
+        return True
+
+    @staticmethod
+    def parse_data_type(data_type_mapping: dict, avro_schema: dict) -> dict:
+        """ Convert data types from avro to json format
+        :param data_type_mapping: mapping from avro to json data types
+        :param avro_schema: schema comes with the avro file
+        :return schema_dict with data types converted from avro to json standards
+        """
+        schema_dict = {}
+        for i in avro_schema["fields"]:
+            data_type = i["type"]
+            # If field is nullable there will be a list of types and we need to make sure to map the whole list according to data_type_mapping
+            if type(data_type) is list:
+                datatype_list = []
+                for dt in data_type:
+                    dt = data_type_mapping[dt]
+                    datatype_list.append(dt)
+                schema_dict[i["name"]] = datatype_list
```

Comment:
can replace all this with a single line list comprehension
```suggestion
                schema_dict[i["name"]] = [data_type_mapping[dtype] for dtype in data_type]
```

Reply:
Applied suggested change